api/v1beta1/safemode.go

```diff
@@ -0,0 +1,18 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package v1beta1
+
+// SafemodeSpec represents a spec with a switch turning safemode on and all safety net switches to turn off
+// IgnoreAll, by default, is defined by the cluster the controller is running on
+// If the use is running on a cluster that has IgnoreAll as false, they will have to manually ignore the other safety nets they do not want considered
+// If IgnoreAll is true by default, the user can change its value
+type SafemodeSpec struct {
```

Comment:
Are these all safe modes that exist by default, and setting these values to `true` disables them? Could we call this `unsafe` in the yaml spec, then, rather than safe?

Reply:
I think its easier from a developers POV to keep it as `safemode`. I can see how it could be confusing from a user's POV. The idea is that everything related to safemode would be in that spec, so if a user wants to turn of a safety net, they would go to the safemode spec where it is currently turned on. If you think it still wouldnt make sense this way, maybe we can reorganize the SafemodeSpec's parameters so it makes more sense?

Reply:
developer == us, here yes? we can call the structs and feature whatever we want, but I think as far as the yaml spec is concerned, we should try to make it very clear to users that they're being unsafe

Reply:
Noted, new commit will make clear that the spec contain unsafe measures rather than safe measures. Resolving for next commit.